staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -265,6 +265,11 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 		location = *req.URL
 		location.Scheme = h.Location.Scheme
 		location.Host = h.Location.Host
+	} else if len(location.RawQuery) == 0 {
```

Comment:
Shouldn't we combine the queries if they exist on both?
Not sure I understand when this would happen and we'd still drop the query and if this is expected by the user.

Reply:
Good call.

At first I thought this might be too difficult and settled on this approach, but on second look it seems pretty simple to handle.  I updated the PR to merge the location and request queries.  ptal.

Reply:
Thanks!
/lgtm